src/reactor.cc

```diff
@@ -356,8 +356,11 @@ void Reactor::Tock() {
   if (retired()) {
     return;
   }
-
-  if (cycle_step >= cycle_time + refuel_time && core.count() == n_assem_core) {
+  
+  // Check that irradiation and refueling periods are over, that 
+  // the core is full and that it has previously been discharged.
```

Comment:
I'm going to propose a wording that is more precise/clear to me.  Feel free to push back and/or suggest some different version.

```suggestion
  // the core is full and that fuel was successfully discharged in this refueling time.
```

Reply:
thanks for this change, it makes it indeed clearer and emphasizes that discharging has to have happened in the same cycle.